wireless_modules/wireless_module.py

```diff
@@ -84,19 +89,26 @@ def run(self, data_rate=1):
             self.mqtt.wait_for_message()
 
             # get millisecond counter
-            start = time.ticks_ms()
+            prev_data_sent = time.ticks_ms()
+            prev_battery_read = time.ticks_ms()
 
             while self.start_publish:
-                data = self._read_sensors()
-                print("-------Publishing--------")
-
-                # compute time difference
-                time_taken = time.ticks_diff(time.ticks_ms(), start) * ms_to_sec
-
+                # Publish the battery voltage of this wireless module if 5 minutes (300 seconds) have elapsed since last
+                # battery read
+                time_since_last_battery_read = time.ticks_diff(time.ticks_ms(), prev_battery_read) * ms_to_sec
+                if time_since_last_battery_read >= battery_data_rate and self.battery is not None:
+                    battery_voltage = self.battery.read()
+                    self.mqtt.publish(self.battery_topic, ujson.dumps(battery_voltage))
```

Comment:
Set `retain=True` so that if we want to check the battery level via MQTT, we don't have to wait until the next publication when subscribing